### PR TITLE
adding RestorePlan to Backup for GKE

### PR DESCRIPTION
**Pull request by neha-vellanki12**
Adding RestorePlan resource to Backup for GKE 
Part of https://github.com/hashicorp/terraform-provider-google/issues/12634

If this PR is for Terraform, I acknowledge that I have:

- [ X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [ X] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [X ] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/develop/run-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X ] Read [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
`google_gke_backup_restore_plan`
```

### CODE REVIEW

File: mmv1/templates/terraform/examples/gkebackup_restoreplan_all_cluster_resources.tf.erb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  location = "us-central1"
+  backup_config {
+    include_volume_data = true
+    include_secrets = true
+    all_namespaces = true
+  }
+}
+
+resource "google_gke_backup_restore_plan" "all_cluster_resources" {
+  name = "rp-<%= ctx[:vars]['name'] %>"
```
Comment (trodge) [inline mmv1/templates/terraform/examples/gkebackup_restoreplan_all_cluster_resources.tf.erb:27]: This resource name should start with `tf-test-` and match the resource name in the yaml file.

**Review by trodge — CHANGES_REQUESTED** [review]
Could you fix the failing tests? I think the issue is mismatched resource names.

Also, could you add the fields identified by the missing test detector to at least one test?

File: mmv1/templates/terraform/examples/gkebackup_restoreplan_all_cluster_resources.tf.erb (hunk 1 above)
Comment (rileykarson) [reply mmv1/templates/terraform/examples/gkebackup_restoreplan_all_cluster_resources.tf.erb:27]: tf-test should be prepended by `<%= ctx[:vars]['name'] %>` automatically as long as there's a hyphen in the `name` value!

Comment (neha-vellanki12) [reply mmv1/templates/terraform/examples/gkebackup_restoreplan_all_cluster_resources.tf.erb]: I'm still seeing missing tests for the IAM binding. I never had to add tests for this when we added BackupPlan resource, are these tests needed?

File: mmv1/products/gkebackup/RestorePlan.yaml [hunk 2]
```diff
@@ -0,0 +1,452 @@
...
+  import_format:
+    [
+      'projects/{{project}}/locations/{{location}}/restorePlans/{{name}}',
+      '{{name}}',
+    ]
+timeouts: !ruby/object:Api::Timeouts
+autogen_async: true
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: 'gkebackup_restoreplan_all_namespaces'
```
Comment (trodge) [inline mmv1/products/gkebackup/RestorePlan.yaml:39]: This first example needs to have `primary_resource_name` for the iam test template to work properly.

**Review by trodge — CHANGES_REQUESTED** [review]

File: mmv1/templates/terraform/examples/gkebackup_restoreplan_second_transformation.tf.erb [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        group_kinds {
+          resource_kind = "Pod"
+          resource_group = ""
+        }
+        json_path = ".metadata[?(@.name == 'nginx')]"
+      }
+      field_actions {
+        op = "COPY"
+        path = "/spec/initContainers/0/env"
+        from_path: "/spec/containers/0/env"
```
Comment (trodge) [inline mmv1/templates/terraform/examples/gkebackup_restoreplan_second_transformation.tf.erb:60]: This should be `from_path =` instead of `from_path:`

**Review by trodge — APPROVED** [review]

**Pull request merged**